src/Agent.Plugins/GitCliManager.cs

```diff
@@ -565,25 +565,27 @@ protected virtual async Task<int> ExecuteGitCommandAsync(AgentTaskPluginExecutio
             string arg = StringUtil.Format($"{command} {options}").Trim();
             context.Command($"git {arg}");
 
-            var processInvoker = new ProcessInvoker(context, disableWorkerCommands: true);
-            processInvoker.OutputDataReceived += delegate (object sender, ProcessDataReceivedEventArgs message)
+            using (var processInvoker = new ProcessInvoker(context, disableWorkerCommands: true))
```

Comment:
process invoker is our api right? is this the right pattern?

Reply:
ProcessInvoker is our API. I am open to better suggestions for a better pattern